### PR TITLE
CodeGen: Allow language keywords in grain behavior and state interfaces

### LINKED ISSUE #8: Code generation does not correctly escape language keywords used as identifiers
Parameters which use language keywords as their name are not correctly escaped by current code generation, which causes compilation failures.

For example, this will cause generated code to fail to compile:
```c#
public interface IReproGrain : IGrainWithIntegerKey
{
  Task Repro(int @event, bool @new, string @class);
}
```

A bunch of early adopters have run into this issue (serves us right for calling parameters "event" :stuck_out_tongue_closed_eyes:)

### PR DESCRIPTION
**Pull request by ReubenBond**
Open to suggestions here, here's a simple fix to the problem of having parameters with C# keywords in either grain interface methods or state class properties.

This simply escapes all parameter names in the generated code on a per-language basis (I haven't tested VB or F# here - can someone else tell me if this is sane?).

Fixes #8

### CODE REVIEW

File: src/ClientGenerator/ClientGenerator.cs [hunk 1]
```diff
@@ -647,7 +647,7 @@ public int RunMain(string[] args)
                 var options = new CodeGenOptions();
                 bool bootstrap = false;             // Used to handle circular dependencies building the runtime
 
-                // STEP 1 : Parse parameters
+                // STEP 1 : Parse parameters;
```
Comment (sergeybykov) [inline src/ClientGenerator/ClientGenerator.cs]: Why semicolon here? :-)

**Comment by ReubenBond** [comment]
I'm not sure, is going back & modifying git history the done thing? I thought it wasn't

**Comment by sergeybykov** [comment]
I'm not sure either. Still learning the rules of the new world.

**Comment by ReubenBond** [comment]
You might be right, I only ever use the most basic git workflow. I'm looking into rebasing: https://help.github.com/articles/about-git-rebase/

**Comment by ReubenBond** [comment]
Okay, I'm clearly having trouble merging them into one commit. Maybe we should close the PR and I will try it again

**Comment by jthelin** [comment]
I find that regular use of `git pull upsteam master` usually does a decent job of keeping your swim-lane branch(s) fairly small and focused.

**Comment by sergeybykov** [comment]
I think I managed to cherry-pick your three essential commits. I'll try to push them. Let's see if I again misunderstood how Git works.

**Comment by ReubenBond** [comment]
Oh, I see. I just opened #63 to fix this, if we can reject this PR and apply that one, that would work

**Comment by sergeybykov** [comment]
I merged the other PR with a single commit.

**Pull request closed without merging**